parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g

```diff
@@ -513,6 +514,13 @@ alterStatementSuffixRenameBranch
     -> ^(TOK_ALTERTABLE_RENAME_BRANCH $sourceBranch $targetBranch)
     ;
 
+alterStatementSuffixReplaceBranch
+@init { gParent.pushMsg("alter table replace branch", state); }
+@after { gParent.popMsg(state); }
+    : KW_REPLACE KW_BRANCH sourceBranch=Identifier KW_WITH KW_SNAPSHOT? (targetBranch=(Identifier | Number))
```

Comment:
https://iceberg.apache.org/docs/nightly/spark-ddl/#alter-table-replace-branch Spark-iceberg `replace branch` syntax has more properties, e.g.:
```
-- REPLACE historical-tag to reference snapshot 4567 and update the retention to 60 days.
ALTER TABLE prod.db.sample REPLACE TAG `historical-tag` AS OF VERSION 4567 RETAIN 60 DAYS
```

Any thought we need also to support these?

Reply:
I will add these. Was checking the spark syntax, they allow only replacing with snapshot id, not with branch?

Reply:
So, I will change the syntax like:
```
ALTER TABLE prod.db.sample REPLACE BRANCH audit-branch
AS OF SYSTEM_VERSION 4567 RETAIN 60 DAYS
```
and for branch
```
ALTER TABLE prod.db.sample REPLACE BRANCH audit-branch
AS OF BRANCH branch-2 RETAIN 60 DAYS
```

I think we are not using version but system_version everywhere.

Reply:
https://github.com/apache/iceberg/pull/6638/files#diff-7bded55cc5d4893d2eb9d899ac095939e7b47814374d21cff159cb7ded9a503e  Check spark implementation, looks like` spark replace `has more properties, e.g. retain, retention..

Reply:
I have copied those from the create branch & pushed a new change, give a check once if the approach looks fine